library/core/src/ptr/mut_ptr.rs

```diff
@@ -2370,6 +2450,17 @@ mod verify {
         ($type:ty, $proof_name:ident, add) => {
             #[kani::proof_for_contract(<*mut $type>::add)]
             pub fn $proof_name() {
+                let mut test_val: $type = kani::any::<$type>();
+                let offset: usize = kani::any();
+                let count: usize = kani::any();                
+                let test_ptr: *mut $type = &mut test_val;
+
+                // For integer, composite, and unit types, 1 is the largest offset that
+                // keeps `ptr_with_offset` within bounds.
```

Comment:
It's missing the key part of the explanation: why? why 1 keeps `ptr_with_offset` for these types?

Reply:
This is a legacy implementation for the proof of contract. In our latest implementation (detailed in PR #166), it has been updated to use the pointer generator.